Formula/hsd.rb

```diff
@@ -0,0 +1,22 @@
+require "language/node"
+
+class Hsd < Formula
+  desc "Handshake Daemon & Full Node"
+  homepage "https://handshake.org"
+  url "https://github.com/handshake-org/hsd/archive/v2.1.2.tar.gz"
+  sha256 "4fc39ee5ca2b91186847a31b3244ae5d57235720ac40812cf752bb687c0833cf"
+
+  depends_on "python" => :build
+  depends_on :xcode => :build
+  depends_on "node@10"
+  depends_on "unbound"
+
+  def install
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    (bin/"hsd").write_env_script libexec/"bin/hsd", :PATH => "#{Formula["node@10"].opt_bin}:$PATH"
+  end
+
+  test do
+    system "#{HOMEBREW_PREFIX}/bin/hsd", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

This will probably also fix your missing file issue.

Reply:
Thank you for your feedback. This program is a blockchain node and by starting it, it will attempt to connect with peers on the network and download a bunch of data. I read the formula cookbook and was initially unsure of how to write a sufficient test. 

One possibility would be to start the application with a particular `--prefix` command, as that will configure the location of the database. The node could be instantly killed and then the presence of that directory should mean that the program was able to run. [geth](https://github.com/Homebrew/homebrew-core/blob/master/Formula/ethereum.rb#L39) (Ethereum) does something similar.

Thank you for the review, will push up an update to fix this.

Reply:
I don't love how my new solution has a race condition and could fail on weaker machines. It now spawns the program, sleeps for a second and then kills the program.

Reply:
Another solution would be to write a JavaScript file to disk that imports `hsd` and assert that it works. Will try that solution.